backend/backend.go

```diff
@@ -313,22 +313,26 @@ func (backend *Backend) createAndAddAccount(
 		if err != nil {
 			return err
 		}
-		if len(configurations) != 1 {
-			// TODO: unified-accounts
-			return errp.New("Watch-only accounts don't support mixed inputs yet")
-		}
-		configuration := configurations[0]
 		accountsConfig := backend.config.AccountsConfig()
 		for _, account := range accountsConfig.Accounts {
-			if account.Configuration.Hash() == configuration.Hash() && account.CoinCode == coin.Code() {
-				return errp.WithStack(ErrAccountAlreadyExists)
+			if account.CoinCode == coin.Code() {
+				// We detect a duplicate account (subaccount in a unified account) if any of the
+				// configurations is already present.
+				for _, config := range account.Configurations {
+					for _, config2 := range configurations {
+						if config.Hash() == config2.Hash() {
+							return errp.WithStack(ErrAccountAlreadyExists)
```

Comment:
What about when `persist` is false. Wouldn't you want to detect an existing account anyway?

Reply:
possibly yes, but if so I can only do it later, since the config is still behind the `getSigningConfigurations()` function which we don't call otherwise. But in the next PRs this function will be gone and we can do it.